### PR TITLE
enable attachments for symbols. fixes SI-5975

### PR DESCRIPTION
**Pull request by lrytz**
follow-up on #817

review by @dragos (only the stuff in Namers / NamesDefaults) and @xeno-by (symbol attachments).

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Namers.scala [hunk 1]
```diff
@@ -52,27 +52,6 @@ trait Namers extends MethodSynthesis {
...
     newNamer(context.makeNewScope(tree, tree.symbol))
 
-  // In the typeCompleter (templateSig) of a case class (resp it's module),
-  // synthetic `copy` (reps `apply`, `unapply`) methods are added. To compute
-  // their signatures, the corresponding ClassDef is needed.
-  // During naming, for each case class module symbol, the corresponding ClassDef
-  // is stored in this map. The map is cleared lazily, i.e. when the new symbol
-  // is created with the same name, the old one (if present) is wiped out, or the
-  // entry is deleted when it is used and no longer needed.
-  private val classOfModuleClass = perRunCaches.newWeakMap[Symbol, WeakReference[ClassDef]]()
```
Comment (dragos) [inline src/compiler/scala/tools/nsc/typechecker/Namers.scala:62]: I can't believe that this map is gone! This has been the source of so many leaks and spurious errors in the IDE... We should open a bottle of champaign and dance to he music of the backspace key!

**Comment by dragos** [comment]
LVGTM

**Comment by lrytz** [comment]
Hi @xeno-by, could you please quickly review the changes related to symbol attachments?

**Comment by xeno-by** [comment]
LGTM. Thanks for the great work!

**Pull request merged**